Sources/Controllers/2.0/OACustomPOIViewController.mm

```diff
@@ -68,39 +99,77 @@ - (void)viewDidLoad
 
     self.tableView.delegate = self;
     self.tableView.dataSource = self;
-    self.tableView.tableHeaderView = [OAUtilities setupTableHeaderViewWithText:OALocalizedString(@"search_poi_types_descr") font:kHeaderViewFont textColor:UIColorFromRGB(color_text_footer) lineSpacing:6.0 isTitle:NO];
+    [self.tableView registerClass:OATableViewCustomHeaderView.class forHeaderFooterViewReuseIdentifier:kHeaderId];
+
+    _searchMode = NO;
+    self.searchBar.delegate = self;
+    UITextField *textField = [self.searchBar valueForKey:@"searchField"];
+    textField.backgroundColor = [UIColor colorWithWhite:1 alpha:1];
+
+    [self updateTextShowButton];
 }
 
 - (void)applyLocalization
 {
-    if (_editMode)
+    [self updateScreenTitle];
+
+    self.saveButton.titleLabel.text = OALocalizedString(@"shared_string_save");
+    self.searchBar.placeholder = _searchMode ? @"" : OALocalizedString(@"shared_string_search");
+}
+
+- (void)updateScreenTitle
+{
+    if (_searchMode)
+        self.titleLabel.text = OALocalizedString(@"shared_string_search");
+    else if (_editMode)
         self.titleLabel.text = _filter.name;
     else
         self.titleLabel.text = OALocalizedString(@"create_custom_poi");
+}
 
-    self.saveButton.titleLabel.text = OALocalizedString(@"shared_string_save");
-    [self updateTextShowButton];
+- (void)updateShowButton:(BOOL)hasSelection
+{
+    self.showButton.backgroundColor = hasSelection ? UIColorFromRGB(color_primary_purple) : UIColorFromRGB(color_route_button_inactive);
+    [self.showButton setTintColor:hasSelection ? UIColor.whiteColor : UIColorFromRGB(color_text_footer)];
+    [self.showButton setTitleColor:hasSelection ? UIColor.whiteColor : UIColorFromRGB(color_text_footer) forState:UIControlStateNormal];
+    [self.showButton setUserInteractionEnabled:hasSelection];
+
+    if (_searchMode)
+    {
+        self.bottomView.hidden = YES;
```

Comment:
Simplify by directly setting the bool to hidden property and using a ternary expression for ```tableBottomConstraint.constant```